### PR TITLE
Revised plugin stack

### PR DESCRIPTION
**Pull request by absalomedia**
This is the updated Plugin functionality rewritten to match how Snapshots work

### CODE REVIEW

File: src/machine.js [hunk 1]
```diff
@@ -212,12 +212,37 @@ Machine.prototype.provision = function (cb) {
     this._generic('provision', [], cb);
 };
 
-Machine.prototype.pluginUpdate = function (cb) {
-    this._generic('plugin update', [], cb);
-};
-
-Machine.prototype.pluginRepair = function (cb) {
-    this._generic('plugin repair', [], cb);
+Machine.prototype.plugin = function () {
```
Comment (edin-m) [inline src/machine.js]: 👍

**Comment by edin-m** [comment]
Do you think the plugin functionality is related to a machine or vagrant itself?
That is, should we write:
`vagrant.plugins().install(...)` or
`var machine = vagrant.init(...); machine.plugins().install(...)`

**Comment by absalomedia** [comment]
The way I read https://www.vagrantup.com/docs/cli/plugin.html it can be either machine based or global. I've included the args variable across each command so it can be done either way

**Comment by edin-m** [comment]
Yeah, but the `plugin()` method will be available only from a `Machine`. Anyways, if you don't plan on updating anymore, I'll merge.

**Comment by absalomedia** [comment]
There won’t be more changes to this branch. I just missed removing the old calls in first commit

**Pull request merged**